PhysicsTools/PatAlgos/python/slimming/customizeMiniAOD_HcalFixLegacy2016.py

```diff
@@ -0,0 +1,134 @@
+    process.reducedEgamma.gsfElectronsPFValMap = cms.InputTag("pfEGammaToCandidateRemapper","electrons")
+    process.reducedEgamma.photonsPFValMap      = cms.InputTag("pfEGammaToCandidateRemapper","photons")
+
+
+    #add bugged conditions to GT for comparison
+    process.GlobalTag.toGet.append(cms.PSet(
+        record = cms.string("HcalRespCorrsRcd"),
+        label = cms.untracked.string("bugged"),
+        tag = cms.string("HcalRespCorrs_v6.3_offline") #to be replaced with proper bugged tag
+        )
+    )
+
+
+    #=== TMP FOR TESTING ONLY
+    #process.load("CondCore.DBCommon.CondDBSetup_cfi")
+    #process.es_pool = cms.ESSource("PoolDBESSource",
+    #                               process.CondDBSetup,
+    #                               timetype = cms.string('runnumber'),
+    #                               toGet = cms.VPSet(
+    #                                            cms.PSet(record = cms.string("HcalRespCorrsRcd"),
+    #                                            tag = cms.string("HcalRespCorrs_2016legacy_fixBadCalib")  #based on HcalRespCorrs_v6.3_offline
+    #                                                     )
+    #                                            ),
+    #                               connect = cms.string('frontier://FrontierProd/CMS_CONDITIONS'),
+    #                               authenticationMethod = cms.untracked.uint32(0)
+    #                               )
+    #process.es_prefer_es_pool = cms.ESPrefer( "PoolDBESSource", "es_pool" )
+    #=== END - TMP FOR TESTING ONLY
+
+
+def addDiscardedPFCandidates(process, inputCollection, verbose=False):
+
+    task = getPatAlgosToolsTask(process)
+
+    process.primaryVertexAssociationDiscardedCandidates = process.primaryVertexAssociation.clone(
+        particles = inputCollection,
+        )
+    task.add(process.primaryVertexAssociationDiscardedCandidates)
+
+    process.packedPFCandidatesDiscarded = process.packedPFCandidates.clone(
+        inputCollection = inputCollection,
+        PuppiNoLepSrc = cms.InputTag(""),
+        PuppiSrc = cms.InputTag(""),
+        secondaryVerticesForWhiteList = cms.VInputTag(),
+        vertexAssociator = cms.InputTag("primaryVertexAssociationDiscardedCandidates","original")
+        )
+    task.add(process.packedPFCandidatesDiscarded)
+
+    addKeepStatement(process, "keep patPackedCandidates_packedPFCandidates_*_*",
+                             ["keep patPackedCandidates_packedPFCandidatesDiscarded_*_*"],
+                              verbose=verbose)
+    # Now make the mixed map for rekeying
+    from PhysicsTools.PatAlgos.slimming.packedPFCandidateRefMixer_cfi import packedPFCandidateRefMixer
+    process.oldPFCandToPackedOrDiscarded = packedPFCandidateRefMixer.clone(
+        pf2pf = cms.InputTag(inputCollection.moduleLabel),
+        pf2packed = cms.VInputTag(cms.InputTag("packedPFCandidates"), cms.InputTag("packedPFCandidatesDiscarded"))
+    )
+    task.add(process.oldPFCandToPackedOrDiscarded)
```

Comment:
are there some expected use cases for this oldPFCandToPackedOrDiscarded module?
There are none in this PR and its data is not saved.
Was this intentional?

Reply:
please respond to this comment

Reply:
I think this is a leftover from the Moriond 2017, where it was needed to save also jets made from the non-clean list of candidates.